spec/defines/attr_spec.rb

```diff
@@ -5,28 +5,86 @@
 
   let(:title) { 'test' }
 
-  let(:params) do
-    {
-      source: 'puppet:///modules/test/path/to/file',
-    }
+  context 'No specific relaxed value' do
+    let(:params) do
+      {
+        source: 'puppet:///modules/test/path/to/file',
+      }
+    end
+
+    it do
+      is_expected.to contain_file('/etc/raddb/mods-config/attr_filter/test')
+        .that_notifies('Service[radiusd]')
+        .that_requires('Group[radiusd]')
+        .that_requires('Package[freeradius]')
+        .with_ensure('present')
+        .with_group('radiusd')
+        .with_mode('0640')
+        .with_owner('root')
+        .with_source('puppet:///modules/test/path/to/file')
+    end
+
+    it do
+      is_expected.to contain_concat__fragment('attr-test')
+        .with_content(%r{^attr_filter filter.test {\n\s+key = "\%{User-Name}"\n\s+filename = \${modconfdir}/\${\.:name}/test\n}})
+        .with_order('20')
+        .with_target('/etc/raddb/mods-available/attr_filter')
+    end
   end
```

Comment:
We can make this a lot simpler and easier to maintain.

- we can have the top level context stay largely the same, except for testing the absence of the relaxed value

```suggestion
  let(:params) do
    {
      source: 'puppet:///modules/test/path/to/file',
    }
  end

  it do
    is_expected.to contain_file('/etc/raddb/mods-config/attr_filter/test')
      .that_notifies('Service[radiusd]')
      .that_requires('Group[radiusd]')
      .that_requires('Package[freeradius]')
      .with_ensure('present')
      .with_group('radiusd')
      .with_mode('0640')
      .with_owner('root')
      .with_source('puppet:///modules/test/path/to/file')
  end

  it do
    is_expected.to contain_concat__fragment('attr-test')
      .with_content(%r{^attr_filter filter.test {\n\s+key = "\%{User-Name}"\n\s+filename = \${modconfdir}/\${\.:name}/test\n}})
      .without_content(%r{^\s+relaxed\s+.*$})
      .with_order('20')
      .with_target('/etc/raddb/mods-available/attr_filter')
  end
```